core/controllers/topics_and_skills_dashboard.py

```diff
@@ -166,6 +166,36 @@ def get(self):
         self.render_json(self.values)
 
 
+class UnassignSkillDataHandler(base.BaseHandler):
+    """Provides data for Unassign skill modal."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+    @acl_decorators.can_access_topics_and_skills_dashboard
+    def get(self, skill_id):
+        """Handles GET requests."""
+        skill_ids_assigned_to_some_topic = (
+            topic_services.get_all_skill_ids_assigned_to_some_topic())
+
+        assigned_topics = {}
+        topics = topic_fetchers.get_all_topics()
+        if skill_id in skill_ids_assigned_to_some_topic:
+            for topic in topics:
+                if skill_id in topic.get_all_skill_ids():
+                    assigned_topics[topic.name] = {
+                        'id': topic.id,
+                        'version': topic.version
+                    }
+                    for subtopic in topic.subtopics:
+                        if skill_id in subtopic.skill_ids:
+                            assigned_topics[topic.name].update(
```

Comment:
Don't modify dicts like this. Get all the info you need and then construct the dict just once.

Also, can a skill ID belong to more than one subtopic? If not, break after you find the correct thing.

Also handle the case when the skill is not assigned to a subtopic.

You should probably have a domain object instead of the dict. Convert it to a dict only at the last minute. Also move more of this logic to the domain layer, the controller layer should not be doing logical calculations.

Reply:
Ok created a domain object. 
Yes skill can not belong to more than one subtopic inside a topic. Added the break statement.
Done.